server/reload.go

```diff
@@ -1100,6 +1133,33 @@ func (s *Server) reloadClientTraceLevel() {
 	}
 }
 
+// reloadClusterName detects cluster membership changes triggered
+// due to a reload where the name changes.
+func (s *Server) reloadClusterName() {
```

Comment:
We need to make sure to send this new cluster name to all solicited leafnodes as well.

Reply:
I believe that this addition does not change the fact that during the reload, s.setClusterName() was invoked with the new name. That function invokes s.updatedSolicitedLeafnodes() which from the look of it will close their connection with a name conflict error code.

Reply:
Yes and this would trigger the deadlock sometimes but branch is rebased so now should work.